src/Docker.DotNet/Models/ServiceFilter.cs

```diff
@@ -26,19 +26,12 @@ public ServiceCreationMode[] Mode
         {
             get
             {
-                if (this["mode"].Length == 0)

```

Comment:
I think we have to be more careful here:
```C#
get
{
    var mode = this["mode"]
    if mode == null
    {
        return null
    }
    var lst = new ServiceCreationMode[mode.Length];
    // ... rest of your code
}
```
```C#
// I think the ?. will work here but if it doesn't add the null assignment check.
set => this["mode"] = value?.Select(m => m.ToString()).ToArray();
```

Reply:
Fixed